### PR TITLE
feat: run workers in installed language runtimes

### LINKED ISSUE #80: Load and run workers with installed runtimes
Now that developers can install new runtimes (#68), `wws` should pick the installed ones and run workers based on it. For that, it will identify the new file extensions to look for files and then, prepare the worker and configure the `WASI` context.

If a language runtime is present in the `.wws.toml` config (see #67) and it's not installed, `wws` will show a warning message indicating they are not installed and some workers won't run.

This will require a new `src/runtimes/runtime/external.rs` struct that properly configures the installed language runtime and the WASI context. All this must come from metadata as developers may create their own set of runtimes.

### PR DESCRIPTION
**Pull request by Angelmmiguel**
This PR configures `wws` to run new workers based on the installed language runtimes. Now, `wws` loads the configuration to detect if there're new file extensions available. Then, it will initiate the runtime for files with those extensions so they can receive requests.

I had to update the different structs that initiates the workers to ensure the `Config` is only load once. 

Regarding how the runtimes are configured, the worker `ExternalRuntime` is in charge of preparing the source code when `wws` starts. Then, it will configure the WASI context on every request. All the required configuration is in the Runtime metadata (see #65)

## Example

```
$ ls
worker.rb  wws

$ wws .
⚙️  Loading routes from: .
🗺  Detected routes:
🚀 Start serving requests at http://127.0.0.1:8080

$ wws runtimes install ruby 3.2.0
⚙️  Fetching data from the repository...
🚀 Installing the runtime...
✅ Done

$ wws . 
⚙️  Loading routes from: .
🗺  Detected routes:
    - http://127.0.0.1:8080/worker
      => ./worker.rb (name: default)
🚀 Start serving requests at http://127.0.0.1:8080
```

It closes #80

### CODE REVIEW

**Review by assambar — APPROVED** [review]
LGTM! Learned a lot while reading this!

File: src/router/files.rs [hunk 1]
```diff
@@ -11,25 +12,43 @@ use wax::{Glob, WalkEntry};
...
     /// Initializes a new files instance. It will detect
     /// relevant resources for WWS like the public folder.
-    pub fn new(root: &Path) -> Self {
+    pub fn new(root: &Path, config: &Config) -> Self {
+        let mut extensions = vec![String::from("js"), String::from("wasm")];
+
+        for repo in &config.repositories {
+            for runtime in &repo.runtimes {
+                for ext in &runtime.extensions {
+                    if !extensions.contains(ext) {
```
[the remark this answers is not in the document]
Comment (Angelmmiguel) [reply src/router/files.rs:32]: That's a good point. Let me add a warning message here 👍

File: src/runtimes/manager.rs [hunk 2]
```diff
@@ -25,9 +29,34 @@ pub fn init_runtime(project_root: &Path, path: &Path) -> Result<Box<dyn Runtime
                 path.to_path_buf(),
             )?)),
             "wasm" => Ok(Box::new(NativeRuntime::new(path.to_path_buf()))),
-            _ => Err(anyhow!(format!(
-                "The '{ext_as_str}' extension does not have an associated runtime"
-            ))),
+            other => {
+                let mut runtime_config = None;
```
[the remark this answers is not in the document]
Comment (Angelmmiguel) [reply src/runtimes/manager.rs]: Done! 👍

**Review by ereslibre — APPROVED** [review]

File: src/config.rs [hunk 3]
```diff
@@ -44,16 +44,7 @@ impl Config {
...
-                name: DEFAULT_REPO_NAME.to_string(),
-                url: DEFAULT_REPO_URL.to_string(),
-                runtimes: Vec::new(),
-            };
-
-            Ok(Self {
-                version: 1,
-                repositories: vec![new_repo],
-            })
+            Ok(Self::default())
```
[the remark this answers is not in the document]
Comment (Angelmmiguel) [reply src/config.rs:47]: After checking the [`Default` docs](https://doc.rust-lang.org/std/default/trait.Default.html), I found this:

> Sometimes, you want to fall back to some kind of default value, and don’t particularly care what it is

In this case, it's important to initialize an empty repo, so I need to implement a custom `default()` method.

**Pull request merged**